### PR TITLE
fix: carwash

### PR DESCRIPTION
**Pull request by mafewtm**
## Description

replace progressbar with lib.progressBar, change to numeric for loop, group corresponding config options into singular table

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My code fits the style guidelines.
- [x] My PR fits the contribution guidelines.

### CODE REVIEW

**Comment by mafewtm** [comment]
linting errors are from other portions of smallresources

File: config.lua [hunk 1]
```diff
@@ -165,25 +163,14 @@ Config.Teleports = {
...
-        ['coords'] = vector3(-699.62, -932.7, 19.01),
+Config.CarWash = {
+    dirtLevel = 0.1, -- threshold for the dirt level to be counted as dirty
+    defaultPrice = 20, -- default price for carwash
+    locations = {
+        [1] = {label = 'Hands Free Carwash', coords = vector3(25.29, -1391.96, 29.33)},
+        [2] = {label = 'Hands Free Carwash', coords = vector3(174.18, -1736.66, 29.35)},
+        [3] = {label = 'Hands Free Carwash', coords = vector3(-74.56, 6427.87, 31.44)},
+        [4] = {label = 'Hands Free Carwash', coords = vector3(1363.22, 3592.7, 34.92)},
+        [5] = {label = 'Hands Free Carwash', coords = vector3(-699.62, -932.7, 19.01)}
```
Comment (Manason) [inline config.lua]: Does this array need explicit indexes? https://manason.github.io/effective-fivem-lua/tables/#imply-array-indices

File: client/carwash.lua [hunk 2]
```diff
@@ -42,21 +48,22 @@ CreateThread(function()
...
-                            if dirtLevel > Config.DirtLevel then
+                            if dirtLevel > Config.CarWash.dirtLevel then
                                 TriggerServerEvent('qb-carwash:server:washCar')
                             else
                                 QBCore.Functions.Notify('The vehicle isn\'t dirty', 'error')
                             end
                         end
                     else
-                        DrawText3Ds(Config.CarWash[k]['coords']['x'], Config.CarWash[k]['coords']['y'], Config.CarWash[k]['coords']['z'], 'The car wash is not available ..')
+                        DrawText3Ds(carWashCoords.x, carWashCoords.y, carWashCoords.z, 'The car wash is not available...')
```
Comment (Manason) [inline client/carwash.lua:65]: Unrelated; we should use the import Draw3DText and remove this local function

**Review by Manason — APPROVED** [review]

File: config.lua (hunk 1 above)
Comment (mafewtm) [reply config.lua]: pushed a commit thats hopefully better

File: client/carwash.lua (hunk 2 above)
Comment (mafewtm) [reply client/carwash.lua:65]: How would you like this done? So I can do it properly in the future?

File: client/carwash.lua [hunk 3]
```diff
@@ -41,17 +41,15 @@ RegisterNetEvent('qb-carwash:client:washCar', function()
...
-        sleep = 1000
+        local sleep = 1000
         if IsPedInAnyVehicle(cache.ped, false) then
-            for i = 1, #Config.CarWash.locations do
-                local carWashCoords = Config.CarWash.locations[i].coords
-                local dist = #(PlayerPos - vector3(carWashCoords.x, carWashCoords.y, carWashCoords.z))
-                if dist <= 7.5 and Driver then
+            for _, carWashCoords in ipairs(Config.CarWash.locations) do
+                local dist = #(playerPos - carWashCoords)
+                if dist <= 7.5 and driver then
```
Comment (Manason) [inline client/carwash.lua]: Should still prefer numeric for loop here https://manason.github.io/effective-fivem-lua/tables/#use-numeric-for-loops-when-iterating-over-an-array

**Review by Manason — APPROVED** [review]

File: client/carwash.lua (hunk 2 above)
Comment (Manason) [reply client/carwash.lua:65]: You don't have to do it in this PR, but can import 'qbx-core:utils' and call the util Draw3DText function https://qbox-docs.vercel.app/resources/core/import

Comment (mafewtm) [reply client/carwash.lua:65]: Might go do all that here shortly

**Pull request merged**